src/Psl/Xml/Internal/issue_level_from_xml_error.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Internal;
+
+use LibXMLError;
+use Psl\Xml\Issue\Level;
+
+/**
+ * @psalm-internal \Psl\Xml
+ * @psalm-suppress MissingThrowsDocblock
+ */
+function issue_level_from_xml_error(LibXMLError $error): Level
+{
+    return Level::fromLevelCode($error->level);
+}
```

Comment:
```suggestion
/**
 * @psalm-internal
 *
 * @psalm-pure
 */
function issue_level_from_xml_error(LibXMLError $error): ?Level
{
    switch($error->level) {
        case LIBXML_ERR_ERROR:
            return Level::error();
        case LIBXML_ERR_FATAL:
            return Level::fatal();
        case LIBXML_ERR_WARNING:
            return Level::warning();
    }
    
    return null;
}
```

Reply:
I gave up on the none ... you win ... :)